core/src/math/activation_functions.cpp

```diff
@@ -0,0 +1,38 @@
+#include "jml/math/activation_functions.hpp"
+#include <cmath>
+
+#include <iostream>
+
+namespace jml {
+
+class FastSigmoid : private ActivationFunction {
```

Comment:
Inside the `.cpp` implementation file you only need to actually paste the bodies of the methods, not the entire class. (Doing so may lead to errors, but don't quote me on that.) See model, vector, and matrix for examples.